cpp/src/unary/math_ops.cu

```diff
@@ -361,9 +376,19 @@ std::unique_ptr<cudf::column> transform_fn(cudf::dictionary_column_view const& i
     output->view(), dictionary::detail::get_indices_type_for_size(output->size()), stream, mr);
 }
 
+template <typename UFN, typename T>
+struct is_supported_type : std::is_arithmetic<T> {};
+
+template <typename T>
+struct is_supported_type<DeviceNegate, T>
+  : std::disjunction<std::is_arithmetic<T>, cudf::is_duration_t<T>> {};
```

Comment:
Decimals should probably be supported, too. I don’t think this will include decimals.

Reply:
Do you mind helping me with this one? I think this is a bit more difficult to support because decimal types are not layout compatible with integer types. See this [comment](https://github.com/rapidsai/cudf/blob/branch-25.02/cpp/include/cudf/utilities/traits.hpp#L512). This is in contrast to duration types which are.

I _think_ this matters because the kernel which calls `thrust::transform` relies on `column_view` iterators. And `column_view` [iterators check that the template parameter is "layout compatible"](https://github.com/rapidsai/cudf/blob/branch-25.02/cpp/include/cudf/column/column_view.hpp#L72).

Reply:
Actually, I think I might just need an implementation of `fixed_point_negate` (ie. negation for fixed point types)

Reply:
I think that seems right.